client/src/components/Search/Search.js

```diff
@@ -215,226 +211,203 @@ class Search extends Component {
         input.select();
       }
     }
-  }
+  };
 
   /**
    * Clear the search term and any advanced search form data.
-   * @param {Object} props
+   * @param {Object} propsArg
    */
-  clearFormData(props) {
-    if (this.state.searchText !== '') {
-      this.setState({ searchText: '' });
+  const clearFormData = (propsArg) => {
+    if (searchText !== '') {
+      setSearchText('');
     }
 
-    const formSchemaUrl = (props && props.formSchemaUrl) || this.props.formSchemaUrl;
-    if (formSchemaUrl) {
-      const identifier = (props && props.identifier) || this.props.identifier;
-      this.props.actions.schema.setSchemaStateOverrides(formSchemaUrl, { fields: [] });
-      this.props.actions.reduxForm.reset(identifier);
+    const formSchemaUrlArg = (propsArg && propsArg.formSchemaUrl) || formSchemaUrl;
+    if (formSchemaUrlArg) {
+      const identifierArg = (propsArg && propsArg.identifier) || identifier;
+      actions.schema.setSchemaStateOverrides(formSchemaUrlArg, { fields: [] });
+      actions.reduxForm.reset(identifierArg);
     }
-  }
+  };
+
+  /**
+   * Wrap up all the data into an object and call the onSearch method provided via the props.
+   * @param {Object} overrides Data to overrides over our existing form data.
+   */
+  const doSearch = (overrides = {}) => {
+    // Data to send to the remote service
+    const searchData = {};
+    const fieldData = getData();
+    Object.entries(fieldData).forEach(([key, value]) => {
+      // Strip any prefix from the key
+      let newKey = key;
+      let newValue = value;
+      if (overrides.hasOwnProperty(key)) {
+        newValue = overrides[key];
+      }
+      if (filterPrefix.length > 0 && key.startsWith(filterPrefix)) {
+        newKey = key.substring(filterPrefix.length);
+      }
+      // Avoid adding prefixed and unprefixed keys, preferring the prefixed
+      if (
+        !filterPrefix.length > 0
+        || key !== name
+        || typeof fieldData[`${filterPrefix}${name}`] === 'undefined'
+      ) {
+        searchData[newKey] = newValue;
+      }
+    });
+    const searchTextData = searchData[name] || '';
+    if (
+      displayState !== DISPLAY.VISIBLE ||
+      initialSearchText !== searchTextData ||
+      searchText !== searchTextData
+    ) {
+      setDisplayState(DISPLAY.VISIBLE);
+      setInitialSearchText(searchTextData);
+      setSearchText(searchTextData);
+    }
+    onSearch(searchData);
+  };
 
   /**
    * Clear a search filter and execute a new search
    * @param {string} key Search filter name to clear
    */
-  clearFormFilter(key) {
-    const tag = this.props.tagData[key];
+  const clearFormFilter = (key) => {
+    const tagItem = tagData[key];
     const clearables = { [key]: undefined };
 
-    const { schemaName, filters } = this.props;
-    this.props.actions.reduxForm.change(schemaName, key, '');
-    this.setOverrides({
-      ...this.props,
+    actions.reduxForm.change(schemaName, key, '');
+    setOverrides({
+      props,
```

Comment:
```suggestion
      ...props,
```
Used to be `...this.props`

Reply:
Fixed